cpp/CMakeLists.txt

```diff
@@ -44,12 +44,16 @@ if(arrow_VERSION_MAJOR STREQUAL "0")
   set(ARROW_SO_VERSION "${ARROW_VERSION_MINOR}")
   set(ARROW_FULL_SO_VERSION "${ARROW_SO_VERSION}.${ARROW_VERSION_PATCH}.0")
 else()
-  message(FATAL_ERROR "Need to implement SO version generation for Arrow 1.0+")
+  message(WARNING "Need to implement SO version generation for Arrow 1.0+, "
+                  "using temporary pre-1.0 numbers")
+  set(ARROW_SO_VERSION "99")
+  set(ARROW_FULL_SO_VERSION "99.0.0")
```

Comment:
@kou Is that ok for you?

Reply:
I would prefer to simply go with https://github.com/apache/arrow/pull/4801